src/pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -591,7 +598,7 @@ def _recursive_unpad(nested: Union[Any, List[Any]], value: Optional[Any] = None)
     return [_recursive_unpad(item, value) for item in nested if item != value]
 
 
-def _recursive_pad(nested: List[Any], fill_value: Optional[Any] = None) -> np.array:
+def _recursive_pad(nested: List[Any], fill_value: Optional[Any] = None) -> Any:
```

Comment:
NumPy docs suggests to use `Any` for object arrays. 💭
Can we add dtype for `result`?

https://numpy.org/doc/stable/reference/typing.html#arraylike

Reply:
Thank you, you got the point. It is not an arraylike. I made it `np.ndarray` in 9d25ea6!